src/typescript/sdk/src/tx-builder.ts

```diff
@@ -0,0 +1,345 @@
+  public removeRecognizedMarkets(
+    marketIds: BCS.Uint64[]
+  ): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::registry::remove_recognized_markets`,
+      type_arguments: [],
+      arguments: [marketIds],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public setRecognizedMarket(marketId: BCS.Uint64): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::registry::set_recognized_market`,
+      type_arguments: [],
+      arguments: [marketId],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public depositFromCoinstore(
+    coin: Types.MoveType,
+    marketId: BCS.Uint64,
+    custodianId: BCS.Uint64,
+    amount: BCS.Uint64
+  ): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::user::deposit_from_coinstore`,
+      type_arguments: [coin],
+      arguments: [marketId, custodianId, amount],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public registerMarketAccount(
+    base: Types.MoveType,
+    quote: Types.MoveType,
+    marketId: BCS.Uint64,
+    custodianId: BCS.Uint64
+  ): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::user::register_market_account`,
+      type_arguments: [base, quote],
+      arguments: [marketId, custodianId],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public registerMarketAccountGenericBase(
+    quote: Types.MoveType,
+    marketId: BCS.Uint64,
+    custodianId: BCS.Uint64
+  ): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::user::register_market_account_generic_base`,
+      type_arguments: [quote],
+      arguments: [marketId, custodianId],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public withdrawToCoinstore(
+    coin: Types.MoveType,
+    marketId: BCS.Uint64,
+    amount: BCS.Uint64
+  ): EconiaTransactionBuilder {
+    const entry: Types.EntryFunctionPayload = {
+      function: `${this.client.econiaAddress}::user::withdraw_to_coinstore`,
+      type_arguments: [coin],
+      arguments: [marketId, amount],
+    };
+    this.entry = entry;
+    return this;
+  }
+
+  public async submitTx(): Promise<Types.Transaction> {
+    if (!this.entry) {
+      throw new Error("No entry function set");
+    }
+    const tx = await this.client.aptosClient.generateTransaction(
+      this.client.userAccount.address(),
+      this.entry
+    );
+    const signedTx = await this.client.aptosClient.signTransaction(
+      this.client.userAccount,
+      tx
+    );
+    const pendingTx = await this.client.aptosClient.submitTransaction(signedTx);
+    const result = await this.client.aptosClient.waitForTransactionWithResult(
+      pendingTx.hash
+    );
+    return result;
+  }
```

Comment:
I suppose this function is the reason `entry` is set as a field, but simply having a `submitTx` function on the client that takes in a `Types.EntryFunctionPayload` argument seems like a much cleaner approach.